corehq/form_processor/models.py

```diff
@@ -1212,11 +1212,11 @@ def relationship(self, relationship):
 
     def __eq__(self, other):
         return isinstance(other, CommCareCaseIndexSQL) and (
-            self.case_id == other.case_id and
-            self.identifier == other.identifier,
-            self.referenced_id == other.referenced_id,
-            self.referenced_type == other.referenced_type,
-            self.relationship_id == other.relationship_id,
+            self.case_id == other.case_id
+            and self.identifier == other.identifier
+            and self.referenced_id == other.referenced_id
+            and self.referenced_type == other.referenced_type
+            and self.relationship_id == other.relationship_id
```

Comment:
This previously returned a tuple of comparison values like `(True, False, True, True)`, which always evaluated to true, meaning all `CommCareCaseIndexSQL` instances compared equal to each other.